### PR TITLE
HW3

### PR DESCRIPTION
**Pull request by w1nway**
### Summary

A brief description of the pull request.

### How it works

Screenshots/screencasts of the pull request introduced functionality.

### Test plan

List of steps to manually test introduced functionality:

- Log in
- Go to http://localhost:3000

### Deploy notes

Notes regarding deployment the contained body of work.
These should note any db migrations, services, scripts, etc.

### Checklist:

- [ ] I have performed a self-review of my own code
- [ ] My pull request is shorter than 500 lines

### CODE REVIEW

File: app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  end
+
+  def create
+    @task = Task.new(task_params)
+    @task.deadline_at = 7.days.after
+
+    if @task.save
+      redirect_to project_tasks_url(@project, @task), notice: "Task was successfully created."
+    else
+      flash.now[:notice] = "Task must have a name."
```
Comment (RasimKhusaenov) [inline app/controllers/tasks_controller.rb:20]: Why? It is incorrect message.

File: app/models/task.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+class Task < ApplicationRecord
+	belongs_to :project
+
+	validates :title, presence: true
+	validates :deadline_at, presence: true
+	validates :project, presence: true
+
+	validate :deadline_at_after_created_at
+	# validates_with MyValidator
```
Comment (RasimKhusaenov) [inline app/models/task.rb:9]: Don't commit commented code

File: app/views/tasks/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  <tbody>
+    <% @tasks.each do |task| %>
+      <tr>
+        <td><%= task.title %></td>
+        <td><%= task.description %></td>
+        <td><%= task.created_at %></td>
+        <td><%= task.deadline_at %></td>
+        <td><%= task.project.name %></td>
+        <td><%= link_to 'Show', url_for([@project, task])%></td>
+        <td><%= link_to 'Edit', edit_project_task_path(:@project, task) %></td>
```
Comment (RasimKhusaenov) [inline app/views/tasks/index.html.erb:26]: Does it work?

File: app/models/task.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+	def deadline_at_after_created_at
+		return if (created_at || Time.current) < deadline_at
+
+		errors.add(:deadline_at, "must be after the date of")
```
Comment (RasimKhusaenov) [inline app/models/task.rb:14]: Poor error message. By error, it will not be very clear what the problem is.

**Review by RasimKhusaenov — COMMENTED** [review]
Almost perfect. It's a pity that the homework was handed over after the deadline. :smiling_face_with_tear:

File: app/models/task.rb (hunk 2 above)
Comment (w1nway) [reply app/models/task.rb:9]: okay

File: app/models/task.rb (hunk 4 above)
Comment (w1nway) [reply app/models/task.rb:14]: okay, I will give correct error messages

**Review by RasimKhusaenov — APPROVED** [review]
Incorrect indents appeared in some files. In the future, let there be an indent of 2 spaces everywhere.

**Pull request closed without merging**